### PR TITLE
Ports: Add ObjFW :^)

### PR DESCRIPTION
**Pull request by bugaevc**
This adds a port of [ObjFW](https://objfw.nil.im/), self-described as _a portable, lightweight framework for the Objective-C language_. The released version is very old and doesn't build for Serenity quite as nicely; a 1.0 release is reportedly upcoming, but for now let's use a Git snapshot (and run `./autogen.sh` before configuring).

https://github.com/SerenityOS/serenity/pull/20158 contains the fixes required to make it build and work. The test suite still doesn't pass, because it tries to use datagram local sockets, which are unimplemented; that being said lots of tests do pass. The included programs (`ofhttp`, `ofdns`, ...) don't quite work either, since they try to sandbox themselves by calling `pledge()`, and our `pledge()` is not a 100% drop in for OpenBSD's (nor should it be), so it gets `EINVAL` on a bad promise name.

This requires enabling Objective-C support in the toolchain. As discussed on Discord, I'm just enabling it for everyone, since it shouldn't add that much more to the toolchain build time.

cc @Midar

### CODE REVIEW

File: Toolchain/BuildGNU.sh
Comment (ADKaster) [inline Toolchain/BuildGNU.sh:302]: This only enables objective C and C++ for gcc, is it already enabled for the clang toolchain?

Comment (BertalanD) [reply Toolchain/BuildGNU.sh:302]: Clang always includes all supported languages.

Comment (BertalanD) [reply Toolchain/BuildGNU.sh:302]: This is a bit cursed, but works:
```sh
❯ export SDKROOT=$(xcrun --show-sdk-path --sdk macosx)
❯ cat test.m
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        // insert code here...
        NSLog(@"Hello, World!");
    }
    return 0;
}
❯ Toolchain/Local/clang/bin/clang --target=arm64-apple-macos14 -framework Foundation test.m
❯ ./a.out
2023-07-27 20:52:10.480 a.out[34111:641191] Hello, World!
```

**Comment by Midar** [comment]
FWIW, ObjFW strongly prefers to be built with Clang, as GCC's ObjC support is severely lacking and buggy. I'm not sure if it's possible for a package to say "I always want to be built with Clang".

**Review by ADKaster — APPROVED** [review]
The port itself looks fine, but we should probably(?) wait until your other PR is merged to pull this one in

**Comment by bugaevc** [comment]
Yep, let's wait for #20158 to land first

**Pull request merged**
